include/aos/common/tools/memory.hpp

```diff
@@ -235,12 +235,14 @@ class SharedPtr : public SmartPtr<T> {
         return *this;
     }
 
+    // cppcheck-suppress-begin noExplicitConstructor
     /**
      * Creates shared pointer from another derived class shared pointer.
      *
      * @param ptr pointer to create from.
      */
     template <typename P>
+    // cppcheck-suppress noExplicitConstructor
```

Comment:
may be it would be better to make the constructor explicit?

Reply:
llvm STL implementation doesn't mark template copy constructor as explicit.

https://github.com/llvm/llvm-project/blob/main/libcxx/include/__memory/shared_ptr.h#L571

Agreed with Oleksandr to follow the same  approach as llvm